compiler/src/dotty/tools/backend/jvm/GenBCode.scala

```diff
@@ -23,10 +23,10 @@ class GenBCode extends Phase { self =>
 
   override def isRunnable(using Context) = super.isRunnable && !ctx.usedBestEffortTasty
 
-  private val superCallsMap = new MutableSymbolMap[Set[ClassSymbol]]
+  private val superCallsMap = new MutableSymbolMap[List[ClassSymbol]]
   def registerSuperCall(sym: Symbol, calls: ClassSymbol): Unit = {
-    val old = superCallsMap.getOrElse(sym, Set.empty)
-    superCallsMap.update(sym, old + calls)
+    val old = superCallsMap.getOrElse(sym, List.empty)
+    superCallsMap.update(sym, old :+ calls)
```

Comment:
What about duplicates?

Reply:
good point! I filtered those out in `BTypesFromSymbols.scala` which is the only place this map seems to be used, but taking care of them here already is neater. updated.